### PR TITLE
2024/01 - Packages update

### PR DESCRIPTION
**Pull request by MaartenBruggink**
> Which issue (if any), is this issue related to?

- Keep our dependencies up-to-date
- Added two extra exports: javascript / typescript without React
- Added a prettier export with minimal rules; tabs/trailing commas
- Changed the lint/prettier scripts for in the repo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,13 +16,16 @@ npm i -D @buildinams/lint
...
 
-- React.js JavaScript (default)
-- Next.js JavaScript (lint/next)
-- React.js TypeScript (lint/react-typescript)
-- Next.js TypeScript (lint/next-typescript)
-- SCSS with CSS Modules (default)
+- JavaScript (@buildinams/lint/eslint/javascript)
+- Typescript (@buildinams/lint/eslint/typescript)
+- React.js JavaScript (@buildinams/lint/eslint/react)
+- Next.js JavaScript (@buildinams/lint/eslint/next)
```
Comment (PauloMFJ) [inline README.md:24]: For consistency should we do this?
```suggestion
- React.js JavaScript (@buildinams/lint/eslint/react-javascript)
- Next.js JavaScript (@buildinams/lint/eslint/next-javascript)
```

File: package.json [hunk 2]
```diff
@@ -1,64 +1,64 @@
...
-    "stylelint": "^14.16.1",
-    "stylelint-config-css-modules": "^4.1.0",
-    "stylelint-config-rational-order": "^0.1.2",
-    "stylelint-config-standard-scss": "^6.1.0",
-    "stylelint-declaration-block-no-ignored-properties": "^2.6.0",
-    "stylelint-order": "^6.0.2"
-  }
+	"name": "@buildinams/lint",
+	"description": "BiA's base lint config(s).",
+	"version": "0.3.0",
```
Comment (PauloMFJ) [inline package.json:4]:
```suggestion
	"version": "0.4.0",
```

File: package.json [hunk 3]
```diff
@@ -1,64 +1,64 @@
...
+		"eslint-plugin-react-hooks": "^4.6.0",
+		"eslint-plugin-simple-import-sort": "^10.0.0",
+		"eslint-plugin-unused-imports": "^3.0.0",
+		"prettier": "^3.1.1",
+		"stylelint": "^16.1.0",
+		"stylelint-config-css-modules": "^4.4.0",
+		"stylelint-config-rational-order": "^0.1.2",
+		"stylelint-config-standard-scss": "^12.0.0",
+		"stylelint-declaration-block-no-ignored-properties": "^2.8.0",
+		"stylelint-order": "^6.0.4"
```
Comment (PauloMFJ) [inline package.json:62]: I had a lot of issues before trying to migrate to the latest Stylelint version. There were a couple of rules that we had enabled that have been depreciated. Let's run this in a project to see if it works fine 🤞

**Review by PauloMFJ — CHANGES_REQUESTED** [review]
Overall looks great :)

File: package.json (hunk 3 above)
Comment (MaartenBruggink) [reply package.json]: Check, I tested it out in the bia-next-template and we had to add a prettier-stylelint module. Now it uses stylelint which extends our prettier conifg; tabs from prettier and import order from stylelint 🎉

Comment (PauloMFJ) [reply package.json]: Nice 👌

File: package.json [hunk 4]
```diff
@@ -1,64 +1,65 @@
...
-    "stylelint": "^14.16.1",
-    "stylelint-config-css-modules": "^4.1.0",
-    "stylelint-config-rational-order": "^0.1.2",
-    "stylelint-config-standard-scss": "^6.1.0",
-    "stylelint-declaration-block-no-ignored-properties": "^2.6.0",
-    "stylelint-order": "^6.0.2"
-  }
+	"name": "@buildinams/lint",
+	"description": "BiA's base lint config(s).",
+	"version": "0.4.0",
```
Comment (PauloMFJ) [inline package.json:4]: You need to re-install after updating version as the version picked up my Github / NPM is based on the package-lock.

**Review by PauloMFJ — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -194,15 +217,12 @@ To then enable format on save in VScode, open your workspace settings (see above
...
-    "editor.defaultFormatter": "stylelint.vscode-stylelint"
-  },
-  "editor.codeActionsOnSave": {
-    "source.fixAll.stylelint": true
-  }
+	"editor.formatOnSave": false,
+	"scss.validate": false,
+	"stylelint.validate": ["scss"],
+	"editor.codeActionsOnSave": {
+		"source.fixAll.stylelint": true
```
Comment (PauloMFJ) [inline README.md:224]: Should this be this now? Same for the other example above.
```suggestion
		"source.fixAll.stylelint": "explicit"
```

**Review by PauloMFJ — APPROVED** [review]

**Review by PauloMFJ — APPROVED** [review]

**Pull request merged**